packages/block-editor/src/store/selectors.js

```diff
@@ -560,18 +560,18 @@ export function getSelectedBlocksInitialCaretPosition( state ) {
 }
 
 /**
- * Returns the current multi-selection set of block client IDs, or an empty
- * array if there is no multi-selection.
+ * Returns the current selection set of block client IDs (multiselection or single selection).
  *
  * @param {Object} state Editor state.
  *
  * @return {Array} Multi-selected block client IDs.
  */
-export const getMultiSelectedBlockClientIds = createSelector(
+export const getSelectedBlockClientIds = createSelector(
 	( state ) => {
 		const { start, end } = state.blockSelection;
 		if ( start === end ) {
-			return [];
+			const selectedBlockClientId = getSelectedBlockClientId( state );
```

Comment:
While I think `getSelectedBlockClientId` is a good abstraction, the fact we're already deconstructing the raw values `start` and `end` with some knowledge of their behavior by having reached this body of the condition, makes me wonder if we ought not just use those values directly.

```js
const { start, end } = state.blockSelection;

if ( start === null || end === null ) {
	return [];
}

if ( start === end ) {
	return [ start ];
}

// ...
```